selfhost/typechecker.jakt

```diff
@@ -7643,6 +7643,7 @@ struct Typechecker {
         UnaryOp(expr, op, span) => {
             let checked_expr = match op {
                 Dereference => .typecheck_expression(expr, scope_id, safety_mode, type_hint: None)
+                Negate => .typecheck_expression_and_dereference_if_needed(expr, scope_id, safety_mode, type_hint, span)
```

Comment:
Please make the commit message explain what's being done, not its result ("typechecker: Pass the correct type hint for unary negate" or something like that)